apps/gateway/src/modules/completions.ts

```diff
@@ -3,39 +3,42 @@ import { Elysia, t } from "elysia";
 
 import { dbClient } from "@hebo/shared-api/middlewares/db-client";
 
-import { provider } from "~gateway/middlewares/provider";
-import { getModelType } from "~gateway/utils/get-model-type";
+import {
+  getModelObject,
+  getProviderConfig,
+  pickModel,
+} from "~gateway/middlewares/provider/service";
 import { convertOpenAICompatibleMessagesToModelMessages } from "~gateway/utils/message-converter";
 
 export const completions = new Elysia({
   name: "completions",
   prefix: "/chat/completions",
 })
   .use(dbClient)
-  .use(provider)
   .post(
     "/",
-    async ({ body, dbClient, provider }) => {
+    async ({ body, dbClient }) => {
       const { model, messages, temperature = 1, stream = false } = body;
+      const foundModel = await getModelObject(dbClient, model);
```

Comment:
I don't understand what getModelObject() does vs pickModel(). What's the advantage vs the previous clear separation between provider and model?